src/test/java/org/broadinstitute/hellbender/utils/UtilsUnitTest.java

```diff
@@ -613,6 +613,75 @@ public void testLastIndexOfLastBoundaries() {
         Assert.assertEquals(result, expected);
     }
 
+    @Test
+    public void testLastIndexOfOneMismatchLastBoundaries() {
+        final String reference = "AAAACCCCTTTTGGGG";
+
+        // match right boundary of reference
+        String query = "TGAGG";
+        int result = Utils.lastIndexOfOneMismatch(reference.getBytes(), query.getBytes());
+        int expected = 11;
+        Assert.assertEquals(result, expected);
+
+        // match left boundary of reference
+        query = "AAGAC";
+        result = Utils.lastIndexOfOneMismatch(reference.getBytes(), query.getBytes());
+        expected = 0;
+        Assert.assertEquals(result, expected);
+    }
+
+    @Test
+    public void testLastIndexOfOneMismatchRandom() {
+        final int num_tests = 100;
+        final int referenceLength = 1000;
+        final int queryLength = 100;
+
+        byte [] reference = new byte[referenceLength];
+        byte [] query = new byte[queryLength];
+
+        final Random rng = Utils.getRandomGenerator();
```

Comment:
You should hard-code a random seed in order to render this test deterministic.  There is a chance, albeit a miniscule one, of the query having a single mismatch randomly in an unexpected place.  Now, I realize how small this chance is, but first, it's good practice to have deterministic tests and second, this future-proofs the test.  For example, what if you copied this code with a smaller query length?

Reply:
There already is a seed hard-coded into the random number generator in Utils.getRandomGenerator()

Reply:
In that case I am satisfied.